### ISSUE TITLE
webpack missing module

### ISSUE THREAD

**Issue by aneogiarcadix** [agracio/edge-js#16]
It seems like webpack can't find type definitions for the package, but I'm not sure why it needs any.
I have a server.ts file, which has a simple hello world edge.js proxy function. I bundle the file using webpack, and it is completed successfully. But the bundled file has "webpackMissingModule" error at the import.
bundle.js

> 

> edge = !(function webpackMissingModule() { var e = new Error("Cannot find module ".""); e.code = 'MODULE_NOT_FOUND'; throw e; }());

Please suggest a fix for this.

> Application has thrown an uncaught exception and is terminated:
> Error: Cannot find module "."
> at webpackMissingModule

**Comment by agracio** [comment]
From my experience you cannot webpack `edge` or `edge-js` because it contains binary files that are being looked up by path. Perhaps someone has solution for it. You should set `edge-js` as external and then copy `node_modules/edge-js` to your working directory. 
If you just running it from your project you shouldn't need to copy anything, but for stand-alone deployments you need to make sure that you `node_modules/edge-js`.
webpack config snippet:
```js
externals: {
       'edge-js': 'require("edge-js")'
                },
```
Also make sure that your webpack target is set to **node**, you cannot use edge in a browser app.

**Comment by aneogiarcadix** [comment]
@agracio Thanks a lot 👍 this solved my problem. And thanks for maintaining this node-v8 build. Cheers!

**Comment by aneogiarcadix** [comment]
There's a follow up problem with this issue. Now that I have included edge-js as external, it is able to run samples with inline code. But when I try to use an assembly file, this error is thrown-
![image](https://user-images.githubusercontent.com/31309583/34702009-6b46f248-f511-11e7-9bdf-baf92ae49e59.png)
This is strange, because I'm able to use the same dll (absolute path to the folder, which also has the .deps file), from another js file(one without any bundling). Is this a known issue? I am using the latest version of the package.

**Comment by agracio** [comment]
This should not happen, I have number of projects both console and Electron that use Webpack and `edge-js`/ `electron-edge-js` as external and call into precompiled dll and have no issues with it. This is a rare error with Edge.Js that usually occurs when something is wrong with node versions.
Could you create a simple Github repo that reproduces the problem and I will take a look at it.

**Comment by aneogiarcadix** [comment]
@agracio I was unable to reproduce the error on a separate project. Everything worked fine.  With edge-js I dont need to write a project.json, right? The deps file should be generated from .csproj . It does get generated, and then edge can't find some specific Nuget packages(Dapper, definitions of SqlDbType in my case), even when the correct version is mentioned in the deps file. Here is an example of the same. https://github.com/aneogiarcadix/EdgeSample

Regarding the error in the previous comment. 
I was able to remove the process.dlopen error by setting EDGE_APP_ROOT to the `bin/debug/netcoreapp2.0` folder. Now that this error was gone, edge started looking for all the dlls, even the ones in `Microsoft.AspNetCore.all`, in this bin folder. I had to paste all those dlls and the errors went away one by one. But then it got stuck again while trying to find libuv.dll, e_sqlite3.dll etc. These are a part of `Microsoft.AspNetCore.all` .

**Comment by agracio** [comment]
There is a problem with .NET Core 2.0 that does not affect 1.0.1+ that I haven't solved yet. I am assuming that you using Linux/macOS as it does not happen on Windows.

**Comment by aneogiarcadix** [comment]
I am using Windows. 
And now I have brand new error to trouble you with. **StackOverflowError**
I removed the dependencies on the `Microsoft.AspNetCore.All` package as in this library I didn't need all of those. Installed only the required packages using Nuget. This removed the previous error related to the .All package.
My code uses `OrderedDictionary`(which implements ` System.Collections.IDictionary`). 
So I set `EDGE_USE_CORECLR=2` and `EDGE_APP_ROOT` to the bin folder, and then this happened.
![image](https://user-images.githubusercontent.com/31309583/35505901-8c63aab4-050d-11e8-8010-ca1c00a32856.png)
Is this related to the unsolved problem? I am using edge-js v9.3.2 and node v8.9.4.
